parser-typechecker/src/Unison/PrettyPrintEnv.hs

```diff
@@ -34,10 +36,43 @@ instance Show PrettyPrintEnv where
   show _ = "PrettyPrintEnv"
 
 fromNames :: Int -> Names -> PrettyPrintEnv
-fromNames length names = PrettyPrintEnv terms' types' where
-  terms' r = safeHead . Set.map HQ'.toHQ $ (Names.termName length r names)
-  types' r = safeHead . Set.map HQ'.toHQ $ (Names.typeName length r names)
-
+fromNames len names = PrettyPrintEnv terms' types' where
+  terms' r = shortestName . Set.map HQ'.toHQ $ (Names.termName len r names)
+  types' r = shortestName . Set.map HQ'.toHQ $ (Names.typeName len r names)
+  shortestName ns = safeHead $ sortOn (length . HQ.toString) (toList ns)
```

Comment:
Important: we pick the shortest name that exists if there are multiple rather than (arbitrarily) picking the first name alphabetically.

Reply:
For `shortestName`, you don't wanna pick `.a.b.c.d` over `computePath` — in the absence of a better metric for semantic closeness, I suggest sorting on the number of segments of the suffix name?  Or I dunno, this is tricky and needs some thought.